File: template.c
#include <stdio.h>
#include <stdlib.h>
#include <getopt.h>

/**
 * Nom du fichier : main.c
 * Description   : [Brève description de ce que fait le programme]
 * Auteur        : [Votre nom]
 * Date de création : [Date de création]
 * Version       : [Version du programme]
 * Licence       : [Licence du programme]
 */

/**
 * Fonction pour traiter les paramètres
 *
 * @param param1 Premier paramètre
 * @param param2 Deuxième paramètre
 */
void process_params(const char *param1, int param2) {
    if (param1 == NULL) {
        fprintf(stderr, "Le paramètre --param1 est obligatoire.\n");
        exit(EXIT_FAILURE);
    }

    if (param2 < 0) {
        fprintf(stderr, "Le paramètre --param2 doit être un nombre positif.\n");
        exit(EXIT_FAILURE);
    }

    // Appel de la fonction principale
    printf("Paramètre 1 : %s\n", param1);
    printf("Paramètre 2 : %d\n", param2);
}

/**
 * Fonction principale
 */
int main(int argc, char *argv[]) {
    int opt;
    char *param1 = NULL;
    int param2 = 0;

    // Définition des options
    while ((opt = getopt(argc, argv, "p:n:")) != -1) {
        switch (opt) {
            case 'p':
                param1 = optarg;
                break;
            case 'n':
                param2 = atoi(optarg);
                break;
            default:
                fprintf(stderr, "Usage: %s -p param1 [-n param2]\n", argv[0]);
                exit(EXIT_FAILURE);
        }
    }

    // Traitement des paramètres
    process_params(param1, param2);

    return EXIT_SUCCESS;
}

File: template.pl
#!/usr/bin/perl

use strict;
use warnings;
use Getopt::Long;
use Pod::Usage;

=head1 NAME

VotreScript.pl - [Brève description de ce que fait le script]

=head1 SYNOPSIS

    VotreScript.pl [options]

    Options:
        --param1      Description du premier paramètre
        --param2      Description du deuxième paramètre
        --help        Affiche la documentation

=head1 DESCRIPTION

[Description détaillée du script, de son objectif et de son fonctionnement]

=head1 OPTIONS

=over 4

=item B<--param1>

Description du premier paramètre

=item B<--param2>

Description du deuxième paramètre

=item B<--help>

Affiche la documentation du script

=back

=head1 EXAMPLES

    perl VotreScript.pl --param1 "Valeur1" --param2 "Valeur2"
    [Description de ce que fait cet exemple]

=head1 AUTHOR

[Votre nom]

=head1 VERSION

[Version du script]

=head1 LICENSE

[Licence du script]

=cut

# Variables pour les paramètres
my $param1;
my $param2;
my $help;

# Lecture des paramètres
GetOptions(
    'param1=s' => \$param1,
    'param2=i' => \$param2,
    'help'     => \$help
) or pod2usage(2);

# Afficher l'aide si nécessaire
pod2usage(1) if $help;

# Validation des paramètres
if (!defined $param1) {
    die "Le paramètre --param1 est obligatoire.\n";
}

if (defined $param2 && $param2 < 0) {
    die "Le paramètre --param2 doit être un nombre positif.\n";
}

# Section : Initialisation
print "Initialisation du script...\n";

# (Ajoutez ici les initialisations nécessaires)

# Section : Fonctions
sub example_function {
    my ($example_param) = @_;
    print "Exécution de example_function avec le paramètre $example_param\n";
    
    # (Ajoutez ici le code de la fonction)
}

# Section : Logique principale du script
print "Exécution du script principal...\n";

# (Ajoutez ici la logique principale du script)
example_function($param1);

# Section : Nettoyage
print "Nettoyage et fin du script...\n";

# (Ajoutez ici le nettoyage si nécessaire)

print "Script terminé avec succès.\n";
exit 0;
